### ISSUE TITLE
empty noderesourcetopology-api repository in kubernetes-sigs

### ISSUE THREAD

**Issue by dims** [kubernetes/community#6308]
### Describe the issue

https://github.com/kubernetes/noderesourcetopology-api/ is totally empty

Looks like instead there is a github.com/k8stopologyawareschedwg/noderesourcetopology-api repository 

Is k8stopologyawareschedwg operated by k8s github management team? 

Looks like `node-feature-discovery` and `scheduler-plugins` uses code from this new repository as well:
https://cs.k8s.io/?q=noderesourcetopology-api&i=nope&files=&excludeFiles=&repos=

Should we cleanup references to `kubernetes/noderesourcetopology-api` from our sigs.yaml / test-infra etc? (and delete the empty repo?)

cc @swatisehgal

**Comment by fromanirh** [comment]
Hi there, commenting as member of the `k8stopologyawareschedwg` group.
I feel that it should be the other way around. AFAIK the plan was to have the node resource topology API merged (in a sufficiently matured/evolved form) into k8s, so we should eventually move the content from github.com/k8stopologyawareschedwg/noderesourcetopology-api  to  github.com/kubernetes/noderesourcetopology-api . I for myself I only learned the latter exists at all in the last couple days. My 2c.

**Comment by mrbobbytables** [comment]
@fromanirh it's the opposite for k8s projects - things that are for the project should be developed inside one of the k8s managed repos. There is significantly more work required for vetting to donate or import an external code base and if the intent is to do that with github.com/k8stopologyawareschedwg/noderesourcetopology-api then we basically have to start from scratch and go through that process.

**Comment by fromanirh** [comment]
> @fromanirh it's the opposite for k8s projects - things that are for the project should be developed inside one of the k8s managed repos. There is significantly more work required for vetting to donate or import an external code base and if the intent is to do that with github.com/k8stopologyawareschedwg/noderesourcetopology-api then we basically have to start from scratch and go through that process.

Thanks for pointing this out. This is unfortunate indeed. In this specific case it should be less painful because that repo holds a CRD definition and code autogenerated out of it.
Anyway, looking forward for other group members (@AlexeyPerevalov @swatisehgal and also @marquiz) to chime in and comment.

**Comment by swatisehgal** [comment]
We had created https://github.com/kubernetes/kubernetes/pull/96275 to populate https://github.com/kubernetes/noderesourcetopology-api/ but at that time we had created this we were proposing changes in the default kube-scheduler and there were concerns around dependency of a native component on a CRD API. In order to move things along and to let the project and the API mature we created: github.com/k8stopologyawareschedwg/noderesourcetopology-api/

Maybe it is time for us to revisit this given that we have [Node feature discovery](https://github.com/kubernetes-sigs/node-feature-discovery) and [NoderesourceTopology scheduler-plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) in the kubernetes org as consumers of this API!

Having said that SIG scheduling had recommended that we move the scheduler plugin to natively in the kube-scheduler and we had started working on this https://github.com/kubernetes/enhancements/pull/2787 but weren't able to pursue this all the way due to other higher priority items.

I think it boils down to our long term plan and timelines, do we want to enable this capability natively in Kubernetes or are happy enough with the current framework? If we want to go with the former then is it still worth populating github.com/k8stopologyawareschedwg/noderesourcetopology-api/ in the interim.

Would be nice to see what others (@AlexeyPerevalov @fromanirh @marquiz @Huang-Wei) think about this.

**Comment by Huang-Wei** [comment]
No matter we will end up with in-tree noderesourcetopology API or not, hosting it under the kubernetes org sounds a good idea. (I also get to know the existence of https://github.com/kubernetes/noderesourcetopology-api/ today...)

**Comment by swatisehgal** [comment]
Apologies @Huang-Wei and @fromanirh, I should have done a better job at keeping you in the loop!

Let's reopen https://github.com/kubernetes/kubernetes/pull/96275 and work on populating github.com/k8stopologyawareschedwg/noderesourcetopology-api then!

Thanks for chiming in on this.

**Comment by swatisehgal** [comment]
@dims We have concluded that we should go ahead and populate https://github.com/kubernetes/noderesourcetopology-api/. Is it okay to close this issue?

**Comment by dims** [comment]
@swatisehgal works! thanks!

/close

**Comment by nikhita** [comment]
/reopen

https://github.com/kubernetes/kubernetes/pull/96275 was closed to inactivity and https://github.com/kubernetes/noderesourcetopology-api/ is still empty.

**Comment by swatisehgal** [comment]
> /reopen
> 
> [kubernetes/kubernetes#96275](https://github.com/kubernetes/kubernetes/pull/96275) was closed to inactivity and https://github.com/kubernetes/noderesourcetopology-api/ is still empty.

Thanks Nikhita for reopening this issue. I didn't get a chance to update https://github.com/kubernetes/kubernetes/pull/96275 as I have been busy but I will update it as soon as I have some time.

/assign

**Comment by nikhita** [comment]
@swatisehgal wanted to follow up on this. https://github.com/kubernetes/noderesourcetopology-api is still empty. Are you still planning to work on this or would you prefer archiving the repo?

I'm also assigning https://github.com/kubernetes/noderesourcetopology-api/issues/1 to you.

**Comment by swatisehgal** [comment]
@nikhita  Thank you so much for following-up on this.
I wasn't able to work on this but certainly plan to do so. I will prioritize this work in the upcoming release cycle.

**Comment by upodroid** [comment]
Forgot to mention this in the meeting but can you look in to moving the repo instead? You can retain the issues, PRs, and releases instead of just moving over the git repo alone.

https://github.com/k8stopologyawareschedwg/noderesourcetopology-api

**Comment by swatisehgal** [comment]
> Forgot to mention this in the meeting but can you look in to [moving](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) the repo instead? You can retain the issues, PRs, and releases instead of just moving over the git repo alone.
> 
> https://github.com/k8stopologyawareschedwg/noderesourcetopology-api

Sorry for the delay in respose. I somehow missed these messages and am following up now.
@upodroid We would be more than happy to donate the existing repo. 

@mrbobbytables I have a [PR](https://github.com/kubernetes/kubernetes/pull/110252) open to bootstrap the repo and was planning to add the API definition as part of a follow up PR but donating the repo altogether shouldn't be a problem from our side.
After reviewing the donation criteria, it looks like we might be able to meet all of them either already or with minor modification to the existing repo. Are there other requirements that we should be aware of to donate the repo? What is the procedure to do so?

**Comment by swatisehgal** [comment]
/remove-lifecycle rotten

We got an [approval](https://github.com/kubernetes/org/issues/4224#issuecomment-1589732129) for the repo to be created yesterday. Once the repo is created, we would be populating it in the upcoming cycles.

**Comment by marquiz** [comment]
I hope we get the repo soon 😊

**Comment by SergeyKanzhelev** [comment]
@swatisehgal do you plan to populate the https://github.com/kubernetes-sigs/noderesourcetopology-api?

**Comment by swatisehgal** [comment]
> @swatisehgal do you plan to populate the https://github.com/kubernetes-sigs/noderesourcetopology-api?

Yes, had pushed a PR: https://github.com/kubernetes-sigs/noderesourcetopology-api/pull/1 for this, waiting for reviews :)

**Comment by MadhavJivrajani** [comment]
/triage accepted
to make the bot happy :)

**Comment by MadhavJivrajani** [comment]
@swatisehgal out of curiosity - what's the status of this?

**Comment by swatisehgal** [comment]
> @swatisehgal out of curiosity - what's the status of this?

I have an open PR: https://github.com/kubernetes-sigs/noderesourcetopology-api/pull/1. It is currently awaiting reviews from API reviewers.

**Comment by dims** [comment]
@swatisehgal  looks like a bot closed that PR :(
